[2023-05-27T10:54:08.008-0500] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_etf_analysis_dag_v_06.stage3_ml_task scheduled__2023-05-27T15:52:24.589556+00:00 [queued]>
[2023-05-27T10:54:08.013-0500] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_etf_analysis_dag_v_06.stage3_ml_task scheduled__2023-05-27T15:52:24.589556+00:00 [queued]>
[2023-05-27T10:54:08.014-0500] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2023-05-27T10:54:08.023-0500] {taskinstance.py:1327} INFO - Executing <Task(SparkSubmitOperator): stage3_ml_task> on 2023-05-27 15:52:24.589556+00:00
[2023-05-27T10:54:08.025-0500] {standard_task_runner.py:57} INFO - Started process 7502 to run task
[2023-05-27T10:54:08.030-0500] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'stock_etf_analysis_dag_v_06', 'stage3_ml_task', 'scheduled__2023-05-27T15:52:24.589556+00:00', '--job-id', '67', '--raw', '--subdir', 'DAGS_FOLDER/assignment/stock-analysis.py', '--cfg-path', '/var/folders/rj/4j_wg7hj631gk16sz5z2gtzc0000gn/T/tmpvh_xf8u1']
[2023-05-27T10:54:08.032-0500] {standard_task_runner.py:85} INFO - Job 67: Subtask stage3_ml_task
[2023-05-27T10:54:08.067-0500] {task_command.py:410} INFO - Running <TaskInstance: stock_etf_analysis_dag_v_06.stage3_ml_task scheduled__2023-05-27T15:52:24.589556+00:00 [running]> on host sujatas-mbp-2
[2023-05-27T10:54:08.107-0500] {taskinstance.py:1545} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='sb' AIRFLOW_CTX_DAG_ID='stock_etf_analysis_dag_v_06' AIRFLOW_CTX_TASK_ID='stage3_ml_task' AIRFLOW_CTX_EXECUTION_DATE='2023-05-27T15:52:24.589556+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-05-27T15:52:24.589556+00:00'
[2023-05-27T10:54:08.111-0500] {base.py:73} INFO - Using connection ID 'spark_stand_alone' for task execution.
[2023-05-27T10:54:08.112-0500] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master local[*] --name arrow-spark --queue root.default ./airflow/dags/assignment/ml_integrate_stage3.py
[2023-05-27T10:54:11.061-0500] {spark_submit.py:490} INFO - data =====================
[2023-05-27T10:54:11.062-0500] {spark_submit.py:490} INFO - (5779, 11)
[2023-05-27T10:54:11.071-0500] {spark_submit.py:490} INFO - Date       Open  ...  vol_moving_avg  adj_close_rolling_med
[2023-05-27T10:54:11.072-0500] {spark_submit.py:490} INFO - 5774  2020-03-26  59.619999  ...    2.263937e+06              65.066000
[2023-05-27T10:54:11.072-0500] {spark_submit.py:490} INFO - 5775  2020-03-27  58.330002  ...    2.252637e+06              64.573000
[2023-05-27T10:54:11.072-0500] {spark_submit.py:490} INFO - 5776  2020-03-30  58.560001  ...    2.277397e+06              64.117000
[2023-05-27T10:54:11.073-0500] {spark_submit.py:490} INFO - 5777  2020-03-31  59.509998  ...    2.284137e+06              63.688333
[2023-05-27T10:54:11.073-0500] {spark_submit.py:490} INFO - 5778  2020-04-01  58.040001  ...    2.334583e+06              63.154666
[2023-05-27T10:54:11.073-0500] {spark_submit.py:490} INFO - 
[2023-05-27T10:54:11.074-0500] {spark_submit.py:490} INFO - [5 rows x 11 columns]
[2023-05-27T10:54:11.896-0500] {spark_submit.py:490} INFO - Traceback (most recent call last):
[2023-05-27T10:54:11.896-0500] {spark_submit.py:490} INFO - File "/Users/sujatabaral/airflow/dags/assignment/ml_integrate_stage3.py", line 51, in <module>
[2023-05-27T10:54:11.897-0500] {spark_submit.py:490} INFO - logging.basicConfig(filename="./airflow/ml-log/metrics.log", level=logging.INFO)
[2023-05-27T10:54:11.897-0500] {spark_submit.py:490} INFO - File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/logging/__init__.py", line 2040, in basicConfig
[2023-05-27T10:54:11.897-0500] {spark_submit.py:490} INFO - h = FileHandler(filename, mode,
[2023-05-27T10:54:11.898-0500] {spark_submit.py:490} INFO - File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/logging/__init__.py", line 1169, in __init__
[2023-05-27T10:54:11.898-0500] {spark_submit.py:490} INFO - StreamHandler.__init__(self, self._open())
[2023-05-27T10:54:11.898-0500] {spark_submit.py:490} INFO - File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/logging/__init__.py", line 1201, in _open
[2023-05-27T10:54:11.899-0500] {spark_submit.py:490} INFO - return open_func(self.baseFilename, self.mode,
[2023-05-27T10:54:11.899-0500] {spark_submit.py:490} INFO - FileNotFoundError: [Errno 2] No such file or directory: '/Users/sujatabaral/airflow/ml-log/metrics.log'
[2023-05-27T10:54:12.069-0500] {spark_submit.py:490} INFO - 23/05/27 10:54:12 INFO ShutdownHookManager: Shutdown hook called
[2023-05-27T10:54:12.072-0500] {spark_submit.py:490} INFO - 23/05/27 10:54:12 INFO ShutdownHookManager: Deleting directory /private/var/folders/rj/4j_wg7hj631gk16sz5z2gtzc0000gn/T/spark-11b4fba5-5726-4e57-b9e8-09ec9c801e3f
[2023-05-27T10:54:12.115-0500] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 421, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local[*] --name arrow-spark --queue root.default ./airflow/dags/assignment/ml_integrate_stage3.py. Error code is: 1.
[2023-05-27T10:54:12.122-0500] {taskinstance.py:1345} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_etf_analysis_dag_v_06, task_id=stage3_ml_task, execution_date=20230527T155224, start_date=20230527T155408, end_date=20230527T155412
[2023-05-27T10:54:12.129-0500] {standard_task_runner.py:104} ERROR - Failed to execute job 67 for task stage3_ml_task (Cannot execute: spark-submit --master local[*] --name arrow-spark --queue root.default ./airflow/dags/assignment/ml_integrate_stage3.py. Error code is: 1.; 7502)
[2023-05-27T10:54:12.183-0500] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2023-05-27T10:54:12.213-0500] {taskinstance.py:2651} INFO - 0 downstream tasks scheduled from follow-on schedule check
